app/models/repository.rb

```diff
@@ -7,17 +7,37 @@ class Repository < ApplicationRecord
 
   scope :only_installer_updates, -> { unscope(where: :installer_updates).where(installer_updates: true) }
   scope :only_mirrored, -> { where(mirroring_enabled: true) }
+  scope :only_custom, -> { where(scc_id: nil) }
+  scope :only_scc, -> { where.not(scc_id: nil) }
 
   validates :name, presence: true
   validates :external_url, presence: true
   validates :local_path, presence: true
 
-  # Mangles remote repo URL to make a nicer local path, see specs for examples
-  def self.make_local_path(url)
-    uri = URI(url)
-    path = uri.path.to_s
-    path.gsub!(%r{^/repo}, '') if (uri.hostname == 'updates.suse.com')
-    path
+  class << self
+
+    def remove_suse_repos_without_tokens!
+      where(auth_token: nil).where('external_url LIKE ?', 'https://updates.suse.com%').delete_all
+    end
+
+    # Mangles remote repo URL to make a nicer local path, see specs for examples
+    def make_local_path(url)
+      uri = URI(url)
+      path = uri.path.to_s
+      path.gsub!(%r{^/repo}, '') if (uri.hostname == 'updates.suse.com')
+      path
+    end
+
+    def by_id(repository_id, custom: false)
+      return Repository.find_by(id: repository_id) if custom
+      Repository.find_by(scc_id: repository_id)
+    end
+
+    def by_url(url)
+      Repository.find_by(external_url: url)
```

Comment:
It's a very simple one-line method and it seems it is used only once. Is it really required?

Reply:
It is used to prevent other classes having knowledge of the database columns.

Reply:
To be honest, I don't think that having knowledge of the DB structure in other places is such a big deal. This column will most likely never change, so it will not need to be refactored. The developer has still to know that the column exists in the DB. They also have to know that this method exists.

Reply:
I like to keep the responsibility of knowing about the internals of the database to as few locations as possible. When your codebase is small, this is not as much of an issue, but as it grows, having database calls everywhere can become a mess and much harder to refactor.

As to not knowing that the method exists, that is a symptom to the problem. It helps if the developer has only one set of places to look.